packages/framework/src/hooks/useRegisterBindings.ts

```diff
@@ -88,6 +88,18 @@ export const useRegisterBindings = <T extends { [key: string]: unknown }>(
     }
   }, [options?.forceState]);
 
+  useEffect(() => {
+    return () => {
+      setWidgetState({
+        values: newValues,
+        invokable: {
+          id: resolvedWidgetId,
+          methods: { ...methods, getId: () => resolvedWidgetId },
+        },
+      });
+    };
+  }, [setWidgetState]);
```

Comment:
This doesn't seem right. This pushes a state update when the component unmounts, which triggers more renders on dependent widgets. We should limit use of setWidgetState.

Update the comparator on line 61 to better handle functions.